src/main/java/mate/academy/springboot/datajpa/repository/specification/requestparam/GetAllProductByCategory.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.springboot.datajpa.repository.specification.requestparam;
+
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.Join;
+import jakarta.persistence.criteria.JoinType;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.repository.specification.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GetAllProductByCategory implements SpecificationProvider<Product> {
+    private static final String FILTER_KEY = "categoryId";
+    private static final String FILTER_NAME = "categoryIn";
+
+    @Override
+    public Specification<Product> getSpecification(String[] params) {
+        return (root, query, cb) -> {
+            Join<Product, Category> products = root.join("category", JoinType.INNER);
+            CriteriaBuilder.In<String> predicate = cb.in(products.get(FILTER_NAME));
```

Comment:
you have constants for words that you use only once and also have the word but without constant. What is the reason for this? Follow the same type of code style everywhere in your program

Reply:
ok